### PR TITLE
added-binary-arithmetic-operators-logical-operators-topk

### PR DESCRIPTION
**Pull request by HarshitNagpal29**
Modified the 6_loadgen.yaml file to include tests for the additional PromQL features mentioned in the issue:

1. Binary Arithmetic Operators (Joins)
2. Logical Operators (and, or, unless)
3. TopK

### CODE REVIEW

**Comment by HarshitNagpal29** [comment]
@bboreham sir, can you kindly approve it so that it can be merged?

**Comment by SuperQ** [comment]
This needs a DCO sign-off. You can use `git commit -s --amend` to add it.

File: prombench/manifests/prombench/benchmark/6_loadgen.yaml [hunk 1]
```diff
@@ -48,6 +48,25 @@ data:
         - expr: histogram_quantile(0.99, sum by(path, le) (rate(codelab_api_request_duration_seconds_bucket{method="POST"}[5m])))
         - expr: histogram_quantile(0.99, sum by(path, method, le) (rate(codelab_api_request_duration_seconds_bucket{method="POST"}[5m])))
         - expr: histogram_quantile(0.99, sum by(instance, le) (rate(codelab_api_request_duration_seconds_bucket{method="POST"}[5m])))
+      - name: binary_arithmetic_operators
+        interval: 15s
+        type: instant
+        queries:
+        - expr: sum(node_memory_MemAvailable_bytes) / sum(node_memory_MemTotal_bytes)
```
Comment (bboreham) [inline prombench/manifests/prombench/benchmark/6_loadgen.yaml:55]: I checked [a running Prombench](http://prombench.prometheus.io/14920/prometheus-release/graph?g0.expr=count(node_memory_MemAvailable_bytes)&g0.tab=1&g0.display_mode=lines&g0.show_exemplars=0&g0.range_input=1h), and there are only 3 series like this; perhaps not great for a benchmark.

File: prombench/manifests/prombench/benchmark/6_loadgen.yaml [hunk 2]
```diff
@@ -48,6 +48,25 @@ data:
...
+        type: instant
+        queries:
+        - expr: rate(node_cpu_seconds_total{mode="system"}[5m]) and rate(node_cpu_seconds_total{mode="user"}[5m])
+        - expr: node_filesystem_avail_bytes unless node_filesystem_size_bytes
+        - expr: rate(node_network_receive_bytes_total[5m]) or rate(node_network_transmit_bytes_total[5m])
+      - name: topk_example
+        interval: 15s
+        type: instant
+        queries:
+        - expr: topk(5, rate(http_requests_total[5m]))
```
Comment (bboreham) [inline prombench/manifests/prombench/benchmark/6_loadgen.yaml:69]: I couldn't find any of these in Prombench.  Where would this metric come from?

**Review by bboreham — COMMENTED** [review]
Hi, thanks for this, it's structurally in the right direction but probably not great choices of metrics.

**Comment by bboreham** [comment]
For reference, I copied this from http://prombench.prometheus.io/14920/prometheus-release/tsdb-status:

<html>
<body>
<h3 style="box-sizing: border-box; margin-bottom: 0.5rem; margin-top: 0px; font-weight: 500; line-height: 1.2; font-size: 1.75rem; color: rgb(33, 37, 41); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Roboto, &quot;Helvetica Neue&quot;, Arial, &quot;Noto Sans&quot;, &quot;Liberation Sans&quot;, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;, &quot;Noto Color Emoji&quot;; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Top 10 series count by metric names</h3>

Name | Count
-- | --
codelab_api_request_duration_seconds_bucket | 5005000
codelab_api_request_duration_seconds_sum | 192500
codelab_api_requests_total | 192500
codelab_api_request_duration_seconds_count | 192500
codelab_api_request_errors_total | 96250
go_gc_duration_seconds | 68780
scrape_samples_post_metric_relabeling | 13759
scrape_samples_scraped | 13759
scrape_duration_seconds | 13759
scrape_series_added | 13759


</body>
</html>

**Comment by HarshitNagpal29** [comment]
@bboreham Hi Sir, hope you are doing well, thanks for your reply, currently I am having my university exams till the end of this month, so not able to look into this matter for now, but as soon as it will end, I will review it properly and make appropriate changes. Hope you understand.

**Comment by bboreham** [comment]
No problem; I think @kushalShukla-web is going to take a look.

File: prombench/manifests/prombench/benchmark/6_loadgen.yaml (hunk 1 above)
Comment (HarshitNagpal2901) [reply prombench/manifests/prombench/benchmark/6_loadgen.yaml:55]: @bboreham Hi Sir, I hope you are doing well. First of all, sorry for the late reply. I'm also really glad that you liked my #743 PR. 

So, now that I'm focusing on this, here are some of the suggested changes from my side. PTAL once so I can create a commit and merge it.

```
- name: binary_arithmetic_operators
  interval: 15s
  type: instant
  queries:
  - expr: sum(rate(codelab_api_requests_total[5m])) / count(codelab_api_requests_total)
  - expr: rate(codelab_api_request_duration_seconds_count[5m]) * 60
  - expr: codelab_api_request_duration_seconds_sum / codelab_api_request_duration_seconds_count
```

As suggested by you, I have made changes, PTAL.

File: prombench/manifests/prombench/benchmark/6_loadgen.yaml (hunk 2 above)
Comment (HarshitNagpal2901) [reply prombench/manifests/prombench/benchmark/6_loadgen.yaml:69]: Some more proposed changes:

```
- name: logical_operators
  interval: 15s
  type: instant
  queries:
  - expr: rate(codelab_api_requests_total{method="GET"}[5m]) and rate(codelab_api_requests_total{method="POST"}[5m])
  - expr: codelab_api_requests_total unless codelab_api_request_errors_total
  - expr: rate(codelab_api_requests_total[5m]) or rate(codelab_api_request_errors_total[5m])
 
- name: topk_example
  interval: 15s
  type: instant
  queries:
  - expr: topk(5, sum by(method, path) (rate(codelab_api_requests_total[5m])))
 ```

**Comment by bboreham** [comment]
This was overtaken by #747. Thanks for your contribution!

**Pull request closed without merging**